package/src/utils/has-integration.ts

```diff
@@ -1,14 +1,32 @@
 import type { HookParameters } from "astro";
+import { AstroError } from "astro/errors";
+
+type HasIntegrationParams =
+	& (
+		| { position?: undefined, relativeTo?: string }
+		| { position: "before" | "after", relativeTo: string }
+	)
+	& {
+		name: string;
+		config: HookParameters<"astro:config:setup">["config"];
+	}
 
 /**
  * Checks whether an integration is already installed.
  *
+ * If `before` is given, returns true only if the integration is installed before the named|provided integration.
+ * If `after` is given, returns true only if the integration is installed after the named|provided integration.
+ *
  * @param {object} params
- * @param {string} params.name
+ * @param {string} params.name - Integration to look up.
+ * @param {undefined | "before" | "after"} params.position - Position in relation to another integration to check.
+ * @param {undefined | string} params.relativeTo - Other integration to check for relative poisition.
  * @param {config} params.config
  *
  * @returns {boolean}
  *
+ * @throws {AstroError} When `params.position` is defined but `params.relativeTo` isn't.
```

Comment:
Is this right? `relativeTo` is an optional param and should fall back to being the current integration?

Reply:
Yeah I don't think not passing `relativeTo` should error. It should just use the current integration as the relative reference

Reply:
It does in the extended hooks. The standalone utility doesn't know what is the current integration.

Reply:
Ah right ok, but with extended hooks ofc this will never not be set. gotcha